### DISCUSSION TITLE
PR for private dependencies

### DISCUSSION THREAD

**Discussion by joedayz** [renovatebot/renovate#17052] [Request Help]
Hi Guys. I am using renovate bot in my GitHub projects with it configuration:

`{
  "extends": [
    "config:base"
  ]   

}
`

These projects in Github using  public and private dependencies. I received the PR when a public dependency is updated in the maven central. 

`        <dependency>
            <groupId>org.apache.httpcomponents</groupId>
            <artifactId>httpclient</artifactId>
            <version>4.5.13</version>
        </dependency>`

But How can I do to update my private dependencies or how I generate a PR when it dependence have a new version?

`        <dependency>
            <groupId>ai.quadim.api</groupId>
            <artifactId>quadim-typelib</artifactId>
            <version>${quadim-typelib-version}</version>
        </dependency>`

I see in the documentation it information:

[docs.renovatebot.com/java/](https://docs.renovatebot.com/java/)

Should I add a config.js in my repo with it configuration? I want to have updates of maven central + updates of my own repositories [mvnrepo.cantara.no/content/reposit...](https://mvnrepo.cantara.no/content/repositories/releases/)

`module.exports = {
    packageRules: [
      {
        matchDatasources: ['maven'],
        registryUrls: ['https://mvnrepo.cantara.no/content/repositories/releases/', 'https://mvnrepo.cantara.no/content/repositories/snapshots/'],
      },
    ],
  };`

Please, give me some advice about this.
Thanks in advance.

Joe

**Comment by rarkins** [comment]
Are you self hosting or using the Mend Renovate app?

**Reply by joedayz** [reply]
I am using the Mend Renovate app.

**Comment by rarkins** [comment]
As you're using the app, you should have your config in a `renovate.json` in the repo. `config.js` is only for self-hosting. If you have not yet merged the onboarding PR then you can make config changes in the onboarding branch/PR.

With the `registryUrls` you have proposed, you will *skip* Maven Central altogether. Is that your intention?

**Reply by joedayz** [reply]
I am using nexus: https://mvnrepo.cantara.no/content/repositories/releases/ , Can I work with nexus?

I publish here my private dependencies and the ones that I don't find will look for them in the central maven.

**Reply by rarkins** [reply]
Will Nexus perform the lookup to Maven Central for you? Because with the rules you proposed above, you are telling Renovate not to look in Maven Central

**Reply by joedayz** [reply]
Sorry @rarkins, I did not trust that if I generated a new version of a core project, it would not generate a PR because it is a dependency of mine that is not published in the central maven.

I generated a tag in a core repo and it released the version 0.43.3. An hour later, all the projects that depended on it were notified with a PR. 

So I haven't had anything to do, and it does what I want.
But, a question came to me:
- Can I tell it to auto-merge from that generated PR only for these core dependencies? 
Because since it's set to auto-merge false, it expects me to approve it, but, I want it to auto merge already for these dependencies that I trust.

**Reply by rarkins** [reply]
Yes, add packageRules with automerge

**Reply by joedayz** [reply]
`{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "digest"],
      "automerge": true
    },
    { 
      "packagePatterns": ["^quadim-typelib", "^quadim-springservicebaseLib"],
      "groupName": "Main branch updates",
      "automerge": true,
      "enabled": true
    }
  ],
  "platformAutomerge": true
}

`

My intention is to automerge only those dependencies and the others that follow with the manual revision of their PR. I'm okay with this?

**Reply by rarkins** [reply]
I don't know what you mean, but you will be merging all non major updates by default, plus major updates of the ones you grouped. Generally not a good idea to automerge major

**Reply by joedayz** [reply]
```
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "updateTypes": ["minor", "patch", "digest"],
      "packagePatterns": ["^quadim-typelib", "^quadim-springservicebaseLib"],
      "groupName": "Main branch updates",
      "automerge": true,
      "enabled": true
    }
  ],
  "platformAutomerge": true
}
```

So I better just declare these packagerules for these dependencies and close my discussion.

**Reply by joedayz** [reply]
Ohhh!! The automerge is not working because already exists an PR open with the change of version.

Update dependency ai.quadim.api:quadim-springservicebaselib to v0.43.3`
`
Is it correct?  @rarkins

**Reply by rarkins** [reply]
That's possibly the cause.
